<?xml version="1.0"?>

<launch>
  <arg name="uri1" default="radio://0/30/2M/E7E7E7E7E2" />
  <arg name="uri2" default="radio://0/70/2M/E7E7E7E7E1" />

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>
  <!-- radio://0/30/2M/E7E7E7E7E2 is newDrone-->
  <group ns="cf1">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri1)" />
      <param name="tf_prefix" value="cf1" />
      <param name="enable_logging" value="True" />
      <param name="enable_logging_imu" value="True" />
      <rosparam>
        genericLogTopics: ["log1"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log1_Variables: ["kalman.stateX", "kalman.stateY", "kalman.stateZ"]
      </rosparam>
    </node>
  </group>
  
  <!-- radio://0/70/2M/E7E7E7E7E1 is RB-->
  <group ns="cf2">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri2)" />
      <param name="tf_prefix" value="cf2" />
      <param name="enable_logging" value="True" />
      <param name="enable_logging_imu" value="True" />
      <rosparam>
        genericLogTopics: ["log2"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log2_Variables: ["kalman.stateX", "kalman.stateY", "kalman.stateZ"]
      </rosparam>
    </node> 
  </group>

    <node name="crazyflie_demo_circle" pkg="crazyflie_demo" type="circDemo.py" output="screen">
    </node>

</launch>